tests/cpgrid/id_entity_entityrep.cpp

```diff
@@ -0,0 +1,386 @@
+//===========================================================================
+/*
+  Copyright 2025 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "config.h"
+
+#define BOOST_TEST_MODULE TemplateOverloadsForIdAndIndexTests
+#include <boost/test/unit_test.hpp>
+
+#include <opm/grid/CpGrid.hpp>
+
+#include <unordered_set>
+
+// This fine attempts to verify that the ids and indices returned by Entity and EntityRep are the same
+// where they are supposed to be the same, and different where they are supposed to be different.
+// This is a test to make sure that modofications to the overloads of id() and index() functions
+// in CpGridData and IndexSet does not break the current behaviour.
+
+struct Fixture
+{
+    Fixture()
+    {
+        int m_argc = boost::unit_test::framework::master_test_suite().argc;
+        char** m_argv = boost::unit_test::framework::master_test_suite().argv;
+        Dune::MPIHelper::instance(m_argc, m_argv);
+        Opm::OpmLog::setupSimpleDefaultLogging();
+    }
+};
+
+BOOST_GLOBAL_FIXTURE(Fixture);
+
+template<typename GridView, typename EntityToIndex, typename EntityRepToIndex>
+bool checkEntityIndex(const GridView& gridView, EntityToIndex&& entityToIndex, EntityRepToIndex&& entityRepToIndex, bool coincide)
+{
+    std::size_t comp_count = 0;
+    std::size_t ref_count = 0;
+    Dune::Hybrid::forEach(std::make_index_sequence<GridView::dimension + 1>{}, [&](auto codim){
```

Comment:
It looks like only comp_count and ref_count are modified inside the body function. Would it make sense to capture just those by reference and the rest by value?